bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/component.jsx

```diff
@@ -0,0 +1,215 @@
+  // This one is kinda tricky, meteor takes sometime to fetch the data and passing down
+  // So the first time its create, the meeting comes as null, sending an error to the client.
+  meeting: {},
+};
+
+const listTransition = {
+  enter: styles.enter,
+  enterActive: styles.enterActive,
+  appear: styles.appear,
+  appearActive: styles.appearActive,
+  leave: styles.leave,
+  leaveActive: styles.leaveActive,
+};
+
+const intlMessages = defineMessages({
+  usersTitle: {
+    id: 'app.userList.usersTitle',
+    description: 'Title for the Header',
+  },
+  messagesTitle: {
+    id: 'app.userList.messagesTitle',
+    description: 'Title for the messages list',
+  },
+  participantsTitle: {
+    id: 'app.userList.participantsTitle',
+    description: 'Title for the Users list',
+  },
+  toggleCompactView: {
+    id: 'app.userList.toggleCompactView.label',
+    description: 'Toggle user list view mode',
+  },
+  ChatLabel: {
+    id: 'app.userList.menu.chat.label',
+    description: 'Save the changes and close the settings menu',
+  },
+  ClearStatusLabel: {
+    id: 'app.userList.menu.clearStatus.label',
+    description: 'Clear the emoji status of this user',
+  },
+  MakePresenterLabel: {
+    id: 'app.userList.menu.makePresenter.label',
+    description: 'Set this user to be the presenter in this meeting',
+  },
+  KickUserLabel: {
+    id: 'app.userList.menu.kickUser.label',
+    description: 'Forcefully remove this user from the meeting',
+  },
+  MuteUserAudioLabel: {
+    id: 'app.userList.menu.muteUserAudio.label',
+    description: 'Forcefully mute this user',
+  },
+  UnmuteUserAudioLabel: {
+    id: 'app.userList.menu.unmuteUserAudio.label',
+    description: 'Forcefully unmute this user',
+  },
+  PromoteUserLabel: {
+    id: 'app.userList.menu.promoteUser.label',
+    description: 'Forcefully promote this viewer to a moderator',
+  },
+  DemoteUserLabel: {
+    id: 'app.userList.menu.demoteUser.label',
+    description: 'Forcefully demote this moderator to a viewer',
+  },
+});
+
+class UserParticipants extends Component {
+
+  componentDidMount() {
+    if (!this.props.compact) {
+      this._usersList.addEventListener('keydown',
+        event => this.props.rovingIndex(event,
+          this._usersList,
+          this._userItems,
+          this.props.users.length));
+    }
+  }
+
+  render() {
+    const {
+      users,
+      currentUser,
+      isBreakoutRoom,
+      intl,
+      makeCall,
+      meeting,
+      getAvailableActions,
+      normalizeEmojiName,
+      isMeetingLocked,
+      compact,
+    } = this.props;
+
+    const userActions =
+      {
+        openChat: {
+          label: () => intl.formatMessage(intlMessages.ChatLabel),
+          handler: (router, user) => router.push(`/users/chat/${user.id}`),
+          icon: 'chat',
+        },
+        clearStatus: {
+          label: () => intl.formatMessage(intlMessages.ClearStatusLabel),
+          handler: user => makeCall('setEmojiStatus', user.id, 'none'),
```

Comment:
I was just looking at this file and thinking that this is not right, and then realized that you were refactoring this part. `makeCall` here is imported in the container, passed down and used directly by the components to interact with our api. This is wrong.
`makeCall` should be imported in the service.js, and then from the service.js you pass down actual handlers, like `setEmojiStatus`, or `assignPresenter`, `kickUser`, `toggleVoice`, `changeRole`, etc.
Could you change that, please?

Reply:
Yes, thanks for the catch, @oswaldoacauan said the same thing for me but i totally forgot to change :+1: